src/components/CurriculumComponents/UnitsTabMobile/UnitsTabMobile.tsx

```diff
@@ -0,0 +1,256 @@
+        $background={"white"}
+        $bottom={["all-spacing-0"]}
+        $right={["all-spacing-0"]}
+        $justifyContent={"center"}
+      >
+        <Button
+          $ma={16}
+          label="Done"
+          data-testid="mobile-done-thread-modal-button"
+          icon="arrow-right"
+          $iconPosition="trailing"
+          iconBackground="black"
+          onClick={handleMobileThreadModal}
+        />
+      </OakFlex>
+    </Box>
+  ) : (
+    <Box $display={["unset", "none"]}>
+      <Box
+        $position={["sticky", "static"]}
+        $top={0}
+        $zIndex={"inFront"}
+        ref={mobileHeaderRef}
+      >
+        <Box
+          $width={"100%"}
+          $background={"white"}
+          $mb={8}
+          data-test-id="filter-mobiles"
+        >
+          <Box>
+            <Box $dropShadow="mobileFilterSelector" $ph={[16, 0]} $pb={16}>
+              <Button
+                label="Highlight a thread"
+                icon="chevron-right"
+                $iconPosition="trailing"
+                variant="buttonStyledAsLink"
+                $mt={16}
+                onClick={handleMobileThreadModal}
+                data-testid="mobile-highlight-thread"
+              />
+              {selectedThread && (
+                <Box
+                  $textOverflow={"ellipsis"}
+                  $whiteSpace={"nowrap"}
+                  $overflow={"hidden"}
+                  data-testid="highlighted-threads-mobile"
+                >
+                  {selectedThread?.title} • {highlightedUnitCount()} units
+                  highlighted
+                </Box>
+              )}
+            </Box>
+            <Box
+              $pt={10}
+              $dropShadow="mobileFilterSelector"
+              $width={"100%"}
+              $ph={[16, 0]}
+              data-testid={"year-selection-mobile"}
+            >
+              <ButtonGroup
+                aria-label="Select a year group"
+                $overflowX={"auto"}
+                $overflowY={"hidden"}
+                $pb={8}
+              >
+                {yearOptions.map((yearOption) => (
+                  <Box key={yearOption} $pt={8} $ml={5}>
+                    <Button
+                      variant="brush"
+                      aria-label={`Year ${yearOption}`}
+                      background={
+                        mobileYearSelection === yearOption ? "black" : "grey20"
+                      }
+                      isCurrent={yearOption === mobileYearSelection}
+                      key={yearOption}
+                      label={`Year ${yearOption}`}
+                      onClick={() => {
+                        setMobileYearSelection(yearOption);
+                        // Scroll into view used also in Lesson Overview - prevents rerender
+                        document
+                          .getElementById(`year-${yearOption}`)
+                          ?.scrollIntoView({ behavior: "smooth" });
```

Comment:
I have to admit the "click to select year" with scroll makes this ticket tricky. Really you shouldn't need to call setMobileYearSelection and leave the state update to the intersection observer, but that means the intersection observer needs a rootMargin to deal with the top spacing and also the years will cycle through all selected options as it navigates (with a delay), which won't look great. 

I think we should chat about this. A single reference could be created and passed into both the CurriculumVisualiser & UnitsTabMobile. CurriculumVisualiser could populate the ref with the item elements and then all of the intersection observer logic could live in the UnitsTabMobile component (which is probably better anyway). It would be easier to deal with this part then, although I still would need to think on it more.